packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -127,7 +128,7 @@ class SharedPreferences {
           .invokeMethod('remove', params)
           .then<bool>((dynamic result) => result);
     } else {
-      _preferenceCache[key] = value;
+      _putToCache(key, value);
```

Comment:
Since this is only used in one place, I don't think we need to split it out into its own method.